### PR TITLE
fix(composite controller): #68 - Skip registration until parent resource has 'Status' subresource enabled

### PR DESCRIPTION
**Pull request by grzesuav**

### CODE REVIEW

File: controller/composite/metacontroller.go [hunk 1]
```diff
@@ -168,10 +168,21 @@ func (mc *Metacontroller) sync(key string) error {
 		}
 		return nil
 	}
+
 	if err != nil {
 		mc.eventRecorder.Eventf(cc, v1.EventTypeNormal, events.ReasonSyncError, "[%s] Sync error - %s", cc.Name, err)
 		return err
 	}
+	parentClient, err := mc.dynClient.Resource(cc.Spec.ParentResource.APIVersion, cc.Spec.ParentResource.Resource)
```
Comment (grzesuav) [inline controller/composite/metacontroller.go:176]: not sure if I can obtain this CRD definition in some better way in this place

File: controller/composite/metacontroller.go [hunk 2]
```diff
@@ -168,10 +168,21 @@ func (mc *Metacontroller) sync(key string) error {
...
+	if err != nil {
+		return err
+	}
+	if hasStatusSubresource := parentClient.APIResource.HasSubresource("status"); !hasStatusSubresource {
+		mc.eventRecorder.Eventf(cc, v1.EventTypeWarning, events.ReasonSyncError, "[%s] Sync error - ignoring, parent resource %s does not have subresource 'Status' enabled", cc.Name, parentClient.GroupVersionKind())
+		klog.InfoS("Ignoring CompositeController", "name", name, "reason", fmt.Sprintf("subresource 'Status' not enabled for %s", parentClient.GroupVersionKind()))
+		// returning, as we cannot do anything until this 'Status' subresource is added to parent resource
+		return nil
```
Comment (AmitKumarDas) [inline controller/composite/metacontroller.go:184]: Can we limit the number of chars in a line to be within 80 chars.
It is unreadable.
In addition, we can use some of the go naming conventions which is quite idiomatic to reduce the line length.
e.g.
```go
if found := parentClient.APIResource.HasSubResource("status"); !found {
  mc.eventRecorder.Eventf(
    cc,
    v1.EventTypeWarning,
    events.ReasonSyncError,
    "[%s] sync error - ignoring ...",
    cc.Name, 
    parentClient.GroupVersionKind(),
  )
  klog.InfoS(
    "Ignoring CompositeController", 
    "name", name, 
    "reason", fmt.Sprintf("subresource 'Status' not enabled for %s", parentClient.GroupVersionKind()),
  )
  return nil
}
```

File: controller/composite/metacontroller.go [hunk 3]
```diff
@@ -168,10 +168,21 @@ func (mc *Metacontroller) sync(key string) error {
...
 		mc.eventRecorder.Eventf(cc, v1.EventTypeNormal, events.ReasonSyncError, "[%s] Sync error - %s", cc.Name, err)
 		return err
 	}
+	parentClient, err := mc.dynClient.Resource(cc.Spec.ParentResource.APIVersion, cc.Spec.ParentResource.Resource)
+	if err != nil {
+		return err
+	}
+	if hasStatusSubresource := parentClient.APIResource.HasSubresource("status"); !hasStatusSubresource {
+		mc.eventRecorder.Eventf(cc, v1.EventTypeWarning, events.ReasonSyncError, "[%s] Sync error - ignoring, parent resource %s does not have subresource 'Status' enabled", cc.Name, parentClient.GroupVersionKind())
+		klog.InfoS("Ignoring CompositeController", "name", name, "reason", fmt.Sprintf("subresource 'Status' not enabled for %s", parentClient.GroupVersionKind()))
```
Comment (AmitKumarDas) [inline controller/composite/metacontroller.go:182]: We should put this under some log level e.g. klog.V(2).InfoS

File: controller/composite/metacontroller.go (hunk 2 above)
Comment (grzesuav) [reply controller/composite/metacontroller.go:184]: sure, will do it

File: controller/composite/metacontroller.go (hunk 3 above)
Comment (grzesuav) [reply controller/composite/metacontroller.go:182]: actually the idea is to make this message verbose, as it is quite important (see related bug)

**Review by AmitKumarDas — APPROVED** [review]

**Pull request merged**